src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +33,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return (double)Math.round(numbers.stream()
+                .mapToInt(n -> (numbers.indexOf(n) % 2 == 1) ? n - 1 : n)
+                .filter(n -> n % 2 == 1)
+                .distinct()
+                .average()
+                .orElseThrow(() -> new NoSuchElementException()));
```

Comment:
```suggestion
                .getAsDouble();
```
as described in common mistakes